### PR TITLE
Html names

### PR DESCRIPTION
**Pull request by epuzanov**
This PR adds support for extended substances names formatting (subscript, superscript, italic, small caps, greek chars).

Names can be entered in 3 different syntax Plain, HTML, LaTeX. Which syntax should be used, can be configured as converterClass for ix.ginas.initializers.StringConverterInitializer.

Default Initializer:
        {
                "class": "ix.ginas.initializers.StringConverterInitializer",
                "converterClass": "ix.ginas.models.converters.PlainStringConverter"
        }

StringConverter class provides 5 main functions for converting Names:

- toHtml- converts name from storage syntax to HTML (used for UI rendering)
- fromHtml - converts name from HTML to storage syntax
- toStd - converts name from storage syntax to Standartiezed Name
- truncate - truncates name
- isValid - check if names syntax are correct 

The JSON "_name" attribute added to Name and SubstanceReference class. This attribute contains HTML representation for the names. And can be used with the new GSRSFrontend.

### CODE REVIEW

File: app/ix/ginas/models/converters/HTMLStringConverter.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        "DELTA", "delta", "EPSILON", "epsilon", "ZETA", "zeta", "ETA", "eta",
+        "THETA", "theta", "IOTA", "iota", "KAPPA", "kappa", "LAMBDA", "lambda",
+        "MU", "mu", "NU", "nu", "XI", "xi", "OMIKRON", "omikron", "PI", "pi",
+        "RHO", "rho", "SIGMA", "sigma", "TAU", "tau", "UPSILON", "upsilon",
+        "PHI", "phi", "CHI", "chi", "PSI", "psi", "OMEGA", "omega", "ss",
+        "Ae", "ae", "Oe", "oe", "Ue", "ue", "Par.", "XFF", "XIII", "XIV",
+        "XV", "XVI", "XVII", "XVIII", "XIX", "XX", "I", "II",
+        "III", "IV", "V", "VI", "VII", "VIII", "IX", "X",
+        "XI", "XII", "D", "L", "N", "-", "", "", "", "", "(", ")", "<", ">", "", "\""};
+    private static Tidy tidy;
```
Comment (dkatzel-ncats) [inline app/ix/ginas/models/converters/HTMLStringConverter.java]: Is JTidy threadsafe?  is there harm making a new instance each time?

Comment (epuzanov) [reply app/ix/ginas/models/converters/HTMLStringConverter.java]: JTidy is not thread-safe. We can make a new tidy instance within validationErrors method.

**Comment by dkatzel-ncats** [comment]
I have made a few changes that will make this String Conversion more configurable.

There is now a new conf property `ix.useHtmlNames` if set to true, then the JSON response will use the HTMLname forthe `_name` value. otherwise it uses the old value

I moved the converter class from a parameter in the initializer to a new conf property `ix.name.converterClass` This will make it easier to switch the conveter class implementation.  Initializers are tricky to override since you provide a list you can't say something like " remove the old StringConverterInitializer and then add this one instead" easily without copying over all the other initializers.   So this way the initializer remains the same and the property for the class name can be changed at will.


This is a good stop gap measure but I don't like that there is only 1 name converter.  Perhaps future versions can support multiple encodings.  We'd have to encode the encoding in the name like with a prefix though something like `{latex} my name in latex format \alpha foo bar`  but we can figure that out later

If this looks good to you I will merge

**Comment by epuzanov** [comment]
Yes, it looks good to me too.

**Comment by dkatzel-ncats** [comment]
OK after discussing with our UI developers it was decided to not break the beta UI that relies on the REST API, I changed the Substance JSON to have both `_name` which returns the old String  and `_name-html` to be able to use both in the UI. I also removed useHtmlName config property since we don't need it anymore

**Review by dkatzel-ncats — APPROVED** [review]
looks good now

**Pull request merged**